Add impl AsRef<[u8]> for Uuid

**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
Add some along this lines:
```
impl AsRef<[u8]> for Uuid {
    #[inline]
    fn as_ref(&self) -> &[u8] {
        &self.as_bytes()[..]
    }
}
```

**Is it blocking?**
No

**Describe alternatives you've considered**
I tried:
```
trait AsBytes { /*...*/ }
impl<T: AsRef<[u8]>> AsBytes for T { /*...*/ }
impl AsBytes for uuid::Uuid { /*...*/ }
```
but got:
```
error[E0119]: conflicting implementations of trait `AsBytes` for type `uuid::Uuid`: ...
note: upstream crates may add a new impl of trait `std::convert::AsRef<[u8]>` for type `uuid::Uuid` in future versions
```
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=a5cd3e47bec30ab4c71ad4270dfb3f04

**Additional context**
It feels this should be implemented for to cover the Interoperability princible.  #191

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -605,6 +605,13 @@ impl Default for Uuid {
     }
 }
 
+impl AsRef<[u8]> for Uuid {
+    #[inline]
+    fn as_ref(&self) -> &[u8] {
+        &self.0
+    }
+}
+
 #[cfg(test)]
 mod tests {
     use crate::{
@@ -1038,9 +1045,12 @@ mod tests {
     fn test_as_bytes() {
         let u = test_util::new();
         let ub = u.as_bytes();
+        let ur = u.as_ref();
 
         assert_eq!(ub.len(), 16);
+        assert_eq!(ur.len(), 16);
         assert!(!ub.iter().all(|&b| b == 0));
+        assert!(!ur.iter().all(|&b| b == 0));
     }
 
     #[test]

